bigquery/google/cloud/bigquery/schema.py

```diff
@@ -43,10 +43,16 @@ def __init__(self, name, field_type, mode='NULLABLE', description=None,
         self.description = description
         self.fields = fields
 
-    def __eq__(self, other):
+    def __key(self):
         return (
-            self.name == other.name and
-            self.field_type.lower() == other.field_type.lower() and
-            self.mode == other.mode and
-            self.description == other.description and
-            self.fields == other.fields)
+            self.name,
+            self.field_type.lower(),
+            self.mode,
+            self.description,
+            self.fields)
```

Comment:
For `RECORD` fields, `self.fields` is a list, which is unhashable:  it needs to be wrapped as a tuple in that case.